### ISSUE TITLE
no-named-as-default-member false positive, with default export with 'default' property

### ISSUE THREAD

**Issue by jquense** [benmosher/eslint-plugin-import#507]
The following produces the incorrect error;

```js
//bar.js
export default { 
  default: 'baz'
};
```

```js
//foo.js
import bar from './bar';

const qux = bar.default;
```
It's incorrectly noting that there is a named export named "default", following the suggestion would lead to this: `import bar, { default } from './bar` which is clearly not right, and I'd suggest not ever what you want to do in this situation. Perhaps that if the property is from the default export, special case allowing 'default' as a property of that export? 

**edit**: actually that would hold true for any named export, with a property of the same name as the export on it.

**Comment by benmosher** [comment]
Your edit is correct. The nature of this rule is not to tell you that you've _definitely_ done something wrong, just that you've matched a pattern that is likely a mistake. 

In your case, it isn't a mistake, but not because it's named `default`. 

I don't know if there is enough of a case for special-casing default versus other names. It would certainly be the only one. 

Let me think about it for a little bit.

**Comment by benmosher** [comment]
Yeah, okay, I think it makes sense to ignore `default`. Seems reasonably unlikely that anyone would be surprised that the `default` export is not a property of the `default` export. 😁
